test/OAuth2/Storage/MongoDBTest.php

```diff
@@ -0,0 +1,120 @@
+        $this->assertFalse($db->checkClientCredentials('oauth_test_client'));
+        $this->assertTrue($db->checkClientCredentials('oauth_test_client', 'testpass'));
+        $this->assertFalse($db->checkClientCredentials('oauth_test_client', 'testpass!!!!'));
+        
+        $db->setClientDetails('oauth_test_client', 'testpass!!!', 'redirect_uri', [], 'map green', 'oauth_test_user_id');
+        $this->assertFalse($db->checkClientCredentials('oauth_test_client', 'testpass'));
+        
+        $db->unsetClientDetails('oauth_test_client');
+        $this->assertFalse($db->checkClientCredentials('oauth_test_client', 'testpass'));
+        
+        $db->setClientDetails('oauth_test_client', 'testpass', 'redirect_uri', [], 'map green', 'oauth_test_user_id');
+        $this->assertTrue($db->checkClientCredentials('oauth_test_client', 'testpass'));
+    }
+    
+    public function testConnection() {
+        $db = new MongoDB('mongodb://localhost:27017/oauth2_server_php');
+        $this->assertNotFalse($db->getClientDetails('oauth_test_client'));
+        $db = new MongoDB(['host' => 'localhost', 'port' => '27017', 'database' => 'oauth2_server_php']);
+        $this->assertNotFalse($db->getClientDetails('oauth_test_client'));
+    }
+
+    public function testIsPublicClient() {
+        $db = new MongoDB('mongodb://localhost:27017/oauth2_server_php');
+        
+        $res = $db->isPublicClient('oauth_test_client');
+        $this->assertFalse($res);
+    }
+    
+    public function testAccessToken() {
+        $db = new MongoDB('mongodb://localhost:27017/oauth2_server_php');
+        
+        $this->assertNotFalse($db->getAccessToken('testtoken'));
+        $db->setAccessToken('testtoken', 'Some Client!!!', 'oauth_test_user_id', 1);
+        $this->assertEquals($db->getAccessToken('testtoken')['client_id'], 'Some Client!!!');
+        $db->unsetAccessToken('testtoken');
+        $this->assertFalse($db->getAccessToken('testtoken'));
+        $db->setAccessToken('testtoken', 'Some Client', 'oauth_test_user_id', 2);
+        $this->assertNotFalse($db->getAccessToken('testtoken'));
+        
+    }
+    
+    public function testAuthorizationCode() {
+        $db = new MongoDB('mongodb://localhost:27017/oauth2_server_php');
+        
+        $this->assertNotFalse($db->getAuthorizationCode('testcode'));
+        $db->setAuthorizationCode('testcode', 'Some Client!!!', 'oauth_test_user_id', 'http://example.com', 0);
+        $this->assertEquals($db->getAuthorizationCode('testcode')['client_id'], 'Some Client!!!');
+        $db->expireAuthorizationCode('testcode');
+        $this->assertFalse($db->getAuthorizationCode('testcode'));
+        $db->setAuthorizationCode('testcode', 'Some Client', 'oauth_test_user_id', 'http://example.com', 23442);
+        $this->assertNotFalse($db->getAuthorizationCode('testcode'));
+    }
+    
+    public function testRefreshToken() {
+        $db = new MongoDB('mongodb://localhost:27017/oauth2_server_php');
```

Comment:
The `MongoDB` object should be passed into the constructor of the storage class.